File: practice/convert/example2_book_library/java/book-library/src/main/java/dev/starryeye/book_library/application/user/query/GetUserService.java
package dev.starryeye.book_library.application.user.query;

import dev.starryeye.book_library.domain.user.User;
import dev.starryeye.book_library.domain.user.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Transactional(readOnly = true)
@Service
@RequiredArgsConstructor
public class GetUserService {

    private final UserRepository userRepository;

    public User get(Long id) {
        return userRepository.findById(id)
                .orElseThrow(() -> new IllegalArgumentException("user is not found, id = " + id));
    }
}

File: practice/convert/example2_book_library/kotlin/book-library/src/main/kotlin/dev/starryeye/book_library/application/user/query/GetUserService.kt
package dev.starryeye.book_library.application.user.query

import dev.starryeye.book_library.application.user.query.result.GetUserResult
import dev.starryeye.book_library.domain.user.UserRepository
import dev.starryeye.book_library.util.findByIdOrThrow
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Transactional(readOnly = true)
@Service
class GetUserService(
    private val userRepository: UserRepository,
) {

    fun getBy(userId: Long): GetUserResult {
        val user = userRepository.findByIdOrThrow(userId, "user is not found, id = $userId")

        return GetUserResult.of(user)
    }
}